Unify decodeEntities use

tomalec opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR does not change any functionality, it just reduces the redundant code.
It removes the custom code for decoding entities and use existing `decodeEntities` from `@wordpress/html-entities`.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run automated tests
2. Smoke test for regression

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Use decode entities from `@wordpress/html-entities`.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

tomalec committed:
Use decodeEntities from `@wordpress/html-entities` instead of custom implementation

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

mikkamp commented on plugins/woocommerce-blocks/packages/prices/utils/test/price.js:
```diff
@@ -5,24 +5,25 @@ import { formatPrice, getCurrency } from '../price';
+		${ 10000000 }       | ${ '€ ' }      | ${ '' }  | ${ ',' }          | ${ '.' }         | ${ 0 }    | ${ '€ 10,000,000' }
+		${ 1000000099 }     | ${ '$' }       | ${ '' }  | ${ ',' }          | ${ '.' }         | ${ 2 }    | ${ '$10,000,000.99' }
+		${ 1000000099 }     | ${ '$' }       | ${ '' }  | ${ '.' }          | ${ ',' }         | ${ 2 }    | ${ '$10.000.000,99' }
+		${ 1000 }           | ${ '&#x62f;' } | ${ '' }  | ${ '.' }          | ${ ',' }         | ${ 2 }    | ${ 'د10,00' }
```
Just wanted to ask a clarifying question as it seems this is the only test line that was added.

Not knowing that character `#x62f` is a RTL character I was expecting a "prefix" to be added to the beginning of the string. Wouldn't we be better off splitting that into two tests so we can confirm each individually:
1. Test that a prefix containing an entity is decoded
2. Test that a RTL entity is placed on the right side of the string

Also do we need to test other entity formats (like `&cent;`) or do we assume that decodeEntities handles all the testing for different variations?

tomalec replied:
Thanks for the suggestions! :) I'll add the other test line (https://github.com/woocommerce/woocommerce/pull/52676/commits/1376cfb14473044f96830d152258c5eb11a52d8b).

> Also do we need to test other entity formats (like &cent;) or do we assume that decodeEntities handles all the testing for different variations?

I think two will be enough, one RTL one LTR. Otherwise, where would we draw the line? [all currency characters](https://github.com/woocommerce/woocommerce/blob/dev/unify-decodeEntity/plugins/woocommerce/tests/e2e-pw/data/settings.js#L10:L10), all possible characters?

tomalec committed:
Add LTR character to decode test

Address https://github.com/woocommerce/woocommerce/pull/52676/files/885af50ca9246bbcd13ef586964399d37b97c0a2#diff-b3804b3092204ca0a05ac2b55aa2c770f02cabb65922f0172cb4305c237bbd46

tomalec committed:
Merge branch 'trunk' into dev/unify-decodeEntity

mikkamp reviewed (approved):
Thanks for the changes here. Since we use ` decodeEntities` in other locations the change looks pretty straight forward to me. I'm having a bit of struggles with some of the tests passing, but those also seem to fail in trunk so most likely not related to the changes in this PR.

I'll go ahead and approve, but do you think it's worth requesting a review from someone on one of the blocks teams?

tomalec committed:
Merge branch 'trunk' into dev/unify-decodeEntity

tomalec committed:
Merge branch 'trunk' into dev/unify-decodeEntity

tomalec commented:
>  I'm having a bit of struggles with some of the tests passing, but those also seem to fail in trunk so most likely not related to the changes in this PR.

I merged the trunk (again), and finally, the tests passed.

> I'll go ahead and approve, but do you think it's worth requesting a review from someone on one of the blocks teams?

I don't think it's needed. It's a rather simple and backward-compatible change
